lib/file_transfer_agent/gcs_util.js

```diff
@@ -137,7 +137,7 @@ function GCSUtil(httpclient, filestream) {
         let matDescKey;
 
         try {
-          if (accessToken) {
+          if (accessToken && !process.env.GOOGLE_APPLICATION_CREDENTIALS) {
```

Comment:
I might be wrong (need to check this), but this still will fail if the Google Library is authenticated in other possible ways.

The library can be authenticated using ([doc](https://cloud.google.com/docs/authentication/application-default-credentials)):
- the env variable
- a global config file
- a google meta server

I don't expect the env variable to be set when two other authentication methods are used.

Reply:
confirming, it still fails when GCP is authenticated using the config file or Google's metadata server. 

Perhaps we could have a configuration toggle which enables the use of Axios instead of gcsClient? So that you don't need to make a guesswork if there can be a conflict with the system-wide setup of GCP.

Reply:
Fixed in next steps